website/docs/concepts/scopes.mdx

```diff
@@ -0,0 +1,331 @@
+internally, and provides a way to access that container to the rest of the widget tree.
+
+```dart
+final valueProvider = StateProvider((ref) => 0);
+
+// DO this
+void main() {
+  runApp(ProviderScope(child: MyApp()));
+}
+
+//DON'T do this:
+final myProviderContainer = ProviderContainer();
+void main(){
+  runApp(MyApp());
+}
+```
+
+:::warning
+Do not use multiple [ProviderContainer]s, without an understanding of how they work. 
+Each will have it's own separate thread of states, which will not be able to access each other.
+Tests are an example of when you might want to use separate [ProviderContainer]s
+in order to make each test's state independent of the others.
+:::
+
+Only create a [ProviderContainer] without a [ProviderScope] for testing and dart-only usage.
+
+## How Riverpod Finds a Provider
+
+When a widget or provider requests the value of a provider, riverpod looks up the state of that provider in the [ProviderContainer] in the nearest 
+[ProviderScope] widget. If neither the provider nor one of it's explicitly listed dependencies is overridden in that scope riverpod continues it's lookup up the widget tree. 
+If the provider has not been overridden in any Widget subtrees the lookup defaults to the [ProviderContainer] in the root [ProviderScope].
+
+Once this process locates the scope in which the provider should reside it determines if the provider has been created yet. 
+If so, it will return the state of the provider.
+However, if the provider has been invalidated or is not currently initialized it will create the state using the provider's build method.
+
+## Scoping Providers to a Widget Subtree
+
+All providers can be scoped to a widget subtree. However, there are some things to keep in mind while doing so. Especially when it comes to understanding how the dependencies property works.
+
+The easiest way to understand subtree scoping is to go through a todo list example with a simple scoped provider,
+and refactor it to use a scoped provider with dependencies.
+
+
+Let's start with a simple TODO data object and a list of TODOs 
+```dart
+/// A simple Todo class
+class Todo {
+  Todo(this.todo);
+  final String todo;
+}
+
+/// Provides a list of [Todo]s.
+final todoListProvider = StateProvider<List<Todo>>(
+    (ref) => [Todo('this'), Todo('that'), Todo('the other')]);
+```
+
+Next we will create a provider to provide an index for the TODO we want to show in a particular widget subtree. 
+We will use this in a `ListView.builder` to set the index.
+```dart
+/// Provides the index for a todo in a particular portion
+/// of the widget tree
+final todoItemIndexProvider = Provider<int>((ref) =>
+    throw UnimplementedError('Todo index not set'));
+
+/// Widget for the whole todo list
+class TodoListWidget extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final length = ref.watch(todoListProvider.select((list) => list.length));
+    print('Build List');
```

Comment:
```suggestion
```

Reply:
This print statement was intended to show the performance optimization.

I'm considering moving the whole example to a page on optimization, and doing an Overlay example (with parent). Also a top-level overrides example with async initialization. Any other examples that you can think of for this page?